src/fenicsxconcrete/sensor.py

```diff
@@ -0,0 +1,67 @@
+
+from pydantic.dataclasses import dataclass
+from typing import Callable
+import numpy as np
+
+
+
+
+@dataclass(config=dict(arbitrary_types_allowed=True))
+class PointSensorDefinition:
+    """
+    Definition of a point sensor
+
+    Args:
+        location: The location of the sensor
+        variable: The variable to be measured
+     
+    """
+    location: tuple[float,float,float]
+    variable: str
+    unit: str
+
+@dataclass(config=dict(arbitrary_types_allowed=True))
+class GlobalSensorDefinition:
+    variable: str
+    unit: str
+
+@dataclass(config=dict(arbitrary_types_allowed=True))
+class PlotDefinition:
+    variable: str
+    #mapping: Callable
+    unit: str
+
+# @dataclass
+# class DolfinXPointSensor:
+#     cells: list[int]
+#     plot_function: df.fem.Function | None
+#     function: df.fem.Function
+#     mapping: Callable
+#     definition: PointSensorDefinition
+
+#     def __init__(self, sensor: SensorDefinition, function: df.fem.FunctionSpace, plot_function: df.fem.Function | None = None):
+#         pass
+
+@dataclass
+class  Sensors:
+    groups: dict[str, list[PointSensorDefinition | GlobalSensorDefinition | PlotDefinition]]
+    #plot_functions: dict[str, df.fem.Function | None]
+    #functions: dict[str, df.fem.Function]
+
```

Comment:
One functionality that is important for me is to be able to create sensors from metadata and export the sensors metadata as well. To export it, that would probably be just a new method here for the class Sensors. To import a sensor from metadata, I guess we would need to write a new function that adds it to the sensors list. That should not be a problem, but means keeping and updating the json schema to the dataclass structure.

Reply:
One of my design goals was that all classes have a very obvious representation as a dictionary/json, and you can just create a dataclass from a dictionary

```python
sensor = Sensor(*dictionary)
```

Since the classes here only describe the sensors and have no methods, serializing and deserializing them should be what you are looking for